include/vdef.h

```diff
@@ -56,6 +56,14 @@
 		assert(ivbprintf >= 0 && ivbprintf < (int)sizeof buf);	\
 	} while (0)
 
+/* Safe strcpy into a fixed-size buffer */
+#define bstrcpy(dst, src)						\
+	do {								\
+		int lbstrcpy = strlen(src);				\
+		assert(lbstrcpy >= 0 && lbstrcpy < sizeof dst);		\
+		strcpy(dst, src);					\
```

Comment:
It didn't occur to me until now that you used `strcpy` here. If we want to ban this family of functions once we have a replacement we can't also use them as-is under the hood.

Reply:
I did actually write the first iteration with `strlcpy()`, but then realized that with the assert, it would not actually provide any benefit over `strcpy()`. Or do you see one?
In my mind, the requirement to not use `strcpy()` directly and use the `bstrcpy()` wrapper would make sure that the assertion is in place to improve safety.
Am I missing something?

Reply:
See the last point in this comment of my draft: https://github.com/varnishcache/varnish-cache/pull/3080#issuecomment-538880822

Reply:
right. So `bstrcpy()` should be an inline function instead, followed by the ban-define.

Reply:
this is getting ugly. Right now it feels like I was on a wrong track.